### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by rcky**
Moves docs from wiki into codebase and adds new information.

### CODE REVIEW

File: docs/loopback/datasources.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Loopback provides connectors which enable your application to consume data from remote services
+using different protocols (such as `loopback-connector-remote`, `loopback-connector-rest` and
+`loopback-connector-soap`).
+
+### Remote vs. Rest
+
+At first sight the remote connector simply seems to be a wrapper for the underlying protocol (REST
+by default). But during the evaluation of the connectors, it seemed to be the case, that the remote
+connector has some important characteristics:
+
```
Comment (lkappeler) [inline docs/loopback/datasources.md]: From the [README](https://github.com/strongloop/loopback-connector-remote) of the loopback-connector-remote

> The remote connector enables you to use a LoopBack application as a data source via REST. You can use the remote connector with a LoopBack application, a Node application, or a browser-based application that uses LoopBack in the client. The connector uses Strong Remoting.
> 
> In general, using the remote connector is more convenient than calling into REST API, and enables you to switch the transport later if you need to.

So may we want to note that it's specifically designed to use loopback apps as a remote datasource

File: docs/loopback/access-control.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Authentication Authorization and Permissions
+
+## Dynamic-Role-Resolver
```
Comment (eventEmitter) [inline docs/loopback/access-control.md]: Did you gain any insights into the inner workings of this? how are those roles resolved? How does it affect performance?

File: docs/loopback/datasources.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   - It seems to optimize the queries to the remote service.
+   - It does not emit the events the other datasources emit (_.e.g._ `before execute`)
+   - It is fully based on strong-remoting
```
Comment (eventEmitter) [inline docs/loopback/datasources.md]: which means that?

File: docs/loopback/mixins.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# Mixins
+
+## General
+
+### Source Code
+
+Loopback Mixins are defined as a function within your mixins folder (`my-mixin.js`):
```
Comment (eventEmitter) [inline docs/loopback/mixins.md]: What are the typical use cases for mixins?

File: docs/loopback/datasources.md (hunk 1 above)
Comment (rcky) [reply docs/loopback/datasources.md]: This is actually what confuses me. The docs seem to imply that it is only a wrapper which allows you to exchange the transport.

**Pull request merged**